fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/Index.java

```diff
@@ -171,9 +172,9 @@ public Index(@Nonnull RecordMetaDataProto.Index proto) throws KeyExpression.Dese
         }
 
         if (proto.hasSubspaceKey()) {
-            subspaceKey = decodeSubspaceKey(proto.getSubspaceKey());
+            setSubspaceKey(decodeSubspaceKey(proto.getSubspaceKey())); // subspace key is explicitly set.
         } else {
-            subspaceKey = name;
+            setSubspaceKey(name);
```

Comment:
I believe spotbugs is kvetching that `subspaceKey` is not initialized because it can't figure out that `setSubspaceKey`, um, sets the subspace key. I *think* we want to disable the check on this constructor.